unit_test/batched/Test_Batched_TeamGemm_Real.hpp

```diff
@@ -1,47 +1,73 @@
+#if defined(HAVE_KOKKOSKERNELS_HALFMATH)
```

Comment:
Note: This may run with `::Test::halfScalarType = float` if half precision is available on only the host or device.

Reply:
whats the difference between test_batched_teamgemm and test_batched_teamgemm_half?

Reply:
the main difference is that test_batched_teamgemm_half always does the following in single precision:
1. generates random numbers
2. checks results on host

since the half precision intrinsic may not be compatible on both host and device